Testing/parsing/test_rule.py

```diff
@@ -70,3 +70,22 @@ def test_bidirectional():
 
     rule_expr = "#! rules\nK(S{u}).B()::cyt => K(S{p})::cyt + B()::cyt @ 3*[K()::cyt]/2*v_1 | 2*[K()::cyt]/3*v_1"
     assert not objects.rules_parser.parse(rule_expr).success
+
+
+def test_replication():
+    rule_expr = "X()::rep =*> X()::rep + X()::rep"
+    assert objects.rule_parser.parse(rule_expr).success
+    # assert objects.rule_parser.parse(rule_expr).data[1] == objects.r1
```

Comment:
Any plans with this commented line?

Reply:
Yes, I had planned to also test the data of parsing, check if the rule is correct. I added tests in commit [796b970](https://github.com/sybila/eBCSgen/pull/107/commits/796b9705f4134af86b738550f4a8d61841aa9e60)